[Linux/Testing] Add namespaces to tests

XToripuru opened the pull request:
#### Feature
Currently tests on Linux use the same network namespace names (app, tool, etc), this PR adds additional identifier (app-N, currently N = 0, etc.). Additionally there's a small code cleanup around tests.

This PR is 1st in a series of PRs to parallelize Linux test execution, unique namespace names are needed for the tests to not interfere which each other.

#### Testing
Linux tests pass locally

andy31415 reviewed (dismissed):
Request changes: this undoes some changes ... I assume bad master merge

andy31415 commented on src/controller/python/matter/clusters/Command.py:
```diff
@@ -69,12 +70,17 @@ def FindCommandClusterObject(isClientSideCommand: bool, path: CommandPath):
-            for _, cmd in inspect.getmembers(getattr(obj, "Commands", None), inspect.isclass):
-                if issubclass(cmd, ClusterCommand) and cmd.command_id == path.CommandId and cmd.is_client == isClientSideCommand:
-                    return cmd
+    for clusterName, obj in inspect.getmembers(sys.modules['matter.clusters.Objects']):
```
this undoes #41975  ... this should not happen

XToripuru committed:
Restore invalid changes

enkiusz committed:
Merge branch 'master' into linux-namespaces

enkiusz committed:
Index assignment removed accidentally during conflict resolution, silence log spam in IPv6 DaD

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Fix submodule commited by mistake, fix __startServer broken while resolving conflicts

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Use DarwinExecutor when running on darwin

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by isort

arkq commented on scripts/tests/chiptest/linux.py:
```diff
@@ -147,48 +149,52 @@ def __init__(self, setup_app_link_up=True, setup_tool_link_up=True,
         for _ in range(100):  # wait at most 10 seconds
             if 'tentative' not in subprocess.check_output(['ip', 'addr'], text=True):
-                log.info("No more tentative addresses")
-                break
```
Why the `break` was removed from here?

arkq commented on scripts/tests/run_test_suite.py:
```diff
@@ -145,12 +149,18 @@ def main(context, dry_run, log_level, target, target_glob, target_skip_glob,
+    if chip_tool is not None:
+        chip_tool = SubprocessInfo(kind='tool', path=chip_tool)
+    else:
+        if not runtime == TestRunTime.MATTER_REPL_PYTHON:
```
Extra indentation not needed here

```suggestion
    elif not runtime == TestRunTime.MATTER_REPL_PYTHON:
        ...
```

arkq reviewed (approved):
Please double check the removed `break` in the `_wait_for_duplicate_address_detection()`.

enkiusz committed:
Post-review changes:

- restore properly working DaD wait loop
- add type hints
- remove redundant indentation

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Merge branch 'master' into linux-namespaces

enkiusz committed:
Apply suggestion from @soares-sergio

Co-authored-by: Sergio Soares <sergiosoares@google.com>

restyled-commits committed:
Restyled by isort
